make/cpp2asm_support.cmake

```diff
@@ -140,26 +140,25 @@ if (MSVC)
   set(CPP_KEEP_WHITESPACE "")
   set(CMAKE_CPP_FLAGS "/nologo")
 else ()
-  # "gcc -E" on a non-.c-extension file gives message:
-  #   "linker input file unused because linking not done"
-  # and doesn't produce any output, so we must use cpp for our .asm files.
-  # we assume it's in the same dir.
-  get_filename_component(compiler_path ${CMAKE_C_COMPILER} PATH)
+  set(CPP_KEEP_COMMENTS -C)
+  set(CPP_NO_LINENUM -P)
+  set(CPP_KEEP_WHITESPACE -traditional-cpp)
+
   # Allow user to set C pre-processor from environment variable
   if (DEFINED ENV{CPP})
     set(CMAKE_CPP $ENV{CPP})
+    set(CMAKE_CPP_FLAGS "")
   else ()
-    find_program(CMAKE_CPP cpp HINTS "${compiler_path}" DOC "path to C preprocessor")
-    if (NOT CMAKE_CPP)
-      message(FATAL_ERROR "cpp is required to build")
-    endif (NOT CMAKE_CPP)
+    # Cmake-discovered compiler on macOS does not include system
+    # directories by default.
```

Comment:
cmake-discovered compiler is `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc` for some reason, which is a compiler without a default `-isysroot`. I.e. if you launch it without an `-isysroot` argument it will not find any macOS system headers at all. This is not the case for `/usr/bin/clang` or `/usr/bin/cc`, which is a host compiler on macOS. I would say it is a bug in cmake.